### PR TITLE
Introduce an ApplicationInstance-specific registry to Boot Cleanup WIP

### PR DESCRIPTION
**Pull request by dgeb**
The ApplicationInstance-specific registry uses its corresponding Application's registry as a failover to resolve registrations and injections.

Soon, this registry will be abstracted behind proxies such as `register` and `inject` on the instance itself.

cc @wycats @tomdale

### CODE REVIEW

File: packages/container/lib/registry.js [hunk 1]
```diff
@@ -18,23 +18,34 @@ var VALID_FULL_NAME_REGEXP = /^[^:]+.+:[^:]+$/;
  @class Registry
 */
 function Registry(options) {
+  this.failover = options && options.failover ? options.failover : null;
```
Comment (tomdale) [inline packages/container/lib/registry.js]: This is a total bikeshed, but I'm not _totally_ in love with the term "failover." To me, a failover implies latch-like behavior: once the "fail" has occurred, you switch everything over to the failover. For example, if the power goes out, you have a "failover" generator that everything switches to.

In this case, it's not really a failover per se… it just looks in its local one first, then expands its search to the parent registry if it's available. Perhaps this terminology is controversial, but I think I'd almost prefer `parentRegistry` to `failover`.

Comment (rwjblue) [inline packages/container/lib/registry.js]: fallback?

Comment (tomdale) [inline packages/container/lib/registry.js]: Fallback seems good.

Comment (dgeb) [inline packages/container/lib/registry.js]: ok, `fallback` it is - thanks @rwjblue!

**Comment by dgeb** [comment]
@tomdale @rwjblue ok, I've fixed up my commits to use `fallback` instead of `failover`

**Comment by tomdale** [comment]
Bless you child. :pray:

**Comment by dgeb** [comment]
:beers:

**Pull request merged**